posthog/temporal/data_imports/pipelines/salesforce/auth.py

```diff
@@ -0,0 +1,41 @@
+import requests
+from django.conf import settings
+
+
+def salesforce_refresh_access_token(refresh_token: str) -> str:
+    res = requests.post(
+        "https://login.salesforce.com/services/oauth2/token",
+        data={
+            "grant_type": "refresh_token",
+            "client_id": settings.SALESFORCE_CONSUMER_KEY,
+            "client_secret": settings.SALESFORCE_CONSUMER_SECRET,
+            "refresh_token": refresh_token,
+        },
+    )
+
+    if res.status_code != 200:
+        err_message = res.json()["error_description"]
+        raise Exception(err_message)
+
+    return res.json()["access_token"]
```

Comment:
A problem I had with hubspot recently is that the access token is short lived (was 30mins) and so we had to refresh this mid-way through pagination if the user had a lot of data (which one user did). 

We likely wanna call refresh at the start of every sync and not rely on the previously stored access token - what is the length of a salesforce token too? We likely will need to add some custom logic into rest_source to handle refresh tokens too

Reply:
Added custom auth